auth-api/src/main/java/io/stargate/auth/api/AuthApiActivator.java

```diff
@@ -17,126 +17,48 @@
   private static final Logger log = LoggerFactory.getLogger(AuthApiActivator.class);
 
-  private BundleContext context;
   private final WebImpl web = new WebImpl();
-  private ServiceReference authenticationServiceReference;
-  private ServiceReference<?> metricsReference;
+  private final ServicePointer<Metrics> metric = ServicePointer.create(Metrics.class);
+  private final ServicePointer<AuthenticationService> authenticationService =
+      ServicePointer.create(
+          AuthenticationService.class,
+          "AuthIdentifier",
+          System.getProperty("stargate.auth_id", "AuthTableBasedService"));
 
-  static String AUTH_IDENTIFIER = System.getProperty("stargate.auth_id", "AuthTableBasedService");
-
-  @Override
-  public void start(BundleContext context) throws Exception {
-    this.context = context;
-    log.info("Starting apiServer....");
-    synchronized (web) {
-      try {
-        String authFilter = String.format("(AuthIdentifier=%s)", AUTH_IDENTIFIER);
-        String metricsFilter = String.format("(objectClass=%s)", Metrics.class.getName());
-        context.addServiceListener(this, String.format("(|%s%s)", authFilter, metricsFilter));
-      } catch (InvalidSyntaxException ise) {
-        throw new RuntimeException(ise);
-      }
-
-      ServiceReference[] refs =
-          context.getServiceReferences(AuthenticationService.class.getName(), null);
-      if (refs != null) {
-        for (ServiceReference ref : refs) {
-          // Get the service object.
-          Object service = context.getService(ref);
-          if (service instanceof AuthenticationService
-              && ref.getProperty("AuthIdentifier") != null
-              && ref.getProperty("AuthIdentifier").equals(AUTH_IDENTIFIER)) {
-            log.info("Setting authenticationService in AuthApiActivator");
-            this.web.setAuthenticationService((AuthenticationService) service);
-            break;
-          }
-        }
-      }
-
-      metricsReference = context.getServiceReference(Metrics.class.getName());
-      if (metricsReference != null) {
-        log.info("Setting metrics in AuthApiActivator");
-        this.web.setMetrics((Metrics) context.getService(metricsReference));
-      }
-
-      if (this.web.getAuthenticationService() != null && this.web.getMetrics() != null) {
-        try {
-          this.web.start();
-          log.info("Started authApiServer....");
-        } catch (Exception e) {
-          log.error("Failed", e);
-        }
-      }
-    }
+  public AuthApiActivator() {
+    super("authApiServer");
   }
 
   @Override
-  public void stop(BundleContext context) {
-    if (authenticationServiceReference != null) {
-      context.ungetService(authenticationServiceReference);
+  protected ServiceAndProperties createService() {
+    web.setAuthenticationService(authenticationService.get());
+    web.setMetrics(metric.get());
+    try {
+      this.web.start();
+    } catch (Exception e) {
+      log.error("Failed", e);
```

Comment:
Could the error message be a bit more verbose?

Reply:
I've ported the original error message. There will be a class name for which this log entry will appear.